emulator/src/kreator/assembler.rs

```diff
@@ -31,97 +45,35 @@ impl Assembler {
 
     pub fn assemble(&self) -> Result<Vec<u8>, &'static str> {
         let label_regex = Regex::new(LABEL_DECL).unwrap();
+        let preprocessed_code = get_preprocessed_code(&self.code)?;
+
         let mut machine_code = Vec::new();
 
-        for line in &self.get_preprocessed_code().unwrap() {
-            let line = label_regex.replace(line, "").to_string();
-            let line = String::from(line.trim());
+        for line in preprocessed_code {
+            let line = label_regex.replace(&line, "").trim().to_string();
+
             if !line.is_empty() {
-                machine_code.extend(to_machine_code(line)?);
+                machine_code.extend(to_machine_code(line.to_string())?);
```

Comment:
Why is this to_string() needed? As I see it to_string() is already called two lines above